packages/plugins/typescript/resolvers/tests/mapping.spec.ts

```diff
@@ -7,23 +7,48 @@ describe('ResolversTypes', () => {
   it('Should build ResolversTypes object when there are no mappers', async () => {
     const result = await plugin(resolversTestingSchema, [], {}, { outputFile: '' });
 
+    expect(result.content).toBeSimilarStringTo(`
+      export type ResolversUnionTypes = {
+        ChildUnion: ( Child ) | ( MyOtherType );
+        MyUnion: ( Omit<MyType, 'unionChild'> & { unionChild?: Maybe<ResolversTypes['ChildUnion']> } ) | ( MyOtherType );
+      };
+    `);
     expect(result.content).toBeSimilarStringTo(`
     export type ResolversTypes = {
       MyType: ResolverTypeWrapper<Omit<MyType, 'unionChild'> & { unionChild?: Maybe<ResolversTypes['ChildUnion']> }>;
       String: ResolverTypeWrapper<Scalars['String']>;
       Child: ResolverTypeWrapper<Child>;
       MyOtherType: ResolverTypeWrapper<MyOtherType>;
-      ChildUnion: ResolversTypes['Child'] | ResolversTypes['MyOtherType'];
+      ChildUnion: ResolverTypeWrapper<ResolversUnionTypes['ChildUnion']>;
```

Comment:
Previously, this is `ResolversTypes['Child'] | ResolversTypes['MyOtherType']` which translate to:

```ts
ResolverTypeWrapper<Child> | ResolverTypeWrapper<MyOtherType>
```

Now, using `ResolverTypeWrapper<ResolversUnionTypes['ChildUnion']>` translate to:

```ts
ResolverTypeWrapper<Child | MyOtherType>
```

I think these 2 are the same semantically. Happy to hear explanation if they are not 🙂